src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -0,0 +1,253 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/CubicScale.hpp"
+
+#include <array>
+#include <pup.h>
+#include <pup_stl.h>
+
+#include "ControlSystem/FunctionOfTime.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Identity.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/RootFinding/CubicEquation.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace CoordMapsTimeDependent {
+
+CubicScale::CubicScale(const double outer_boundary)
+    : outer_boundary_(outer_boundary) {}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::operator()(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  return {
+      {source_coords[0] * (map_list.at(f_of_t_a_).func(time)[0][0] +
+                           (map_list.at(f_of_t_b_).func(time)[0][0] -
+                            map_list.at(f_of_t_a_).func(time)[0][0]) *
+                               square(source_coords[0] / outer_boundary_))}};
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::inverse(
+    const std::array<T, 1>& target_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  // the original coordinates are found by solving for the roots
+  // of (b-a)/X^2*\xi^3 + a*\xi - x = 0, where a and b are the FunctionsOfTime,
+  // X is the outer_boundary, and x represents the mapped coordinates
+  const double cubic_coef_a = (map_list.at(f_of_t_b_).func(time)[0][0] -
+                               map_list.at(f_of_t_a_).func(time)[0][0]) /
+                              square(outer_boundary_);
+  const auto roots = real_cubic_roots(cubic_coef_a, 0.0,
+                                      map_list.at(f_of_t_a_).func(time)[0][0],
+                                      -target_coords[0]);
+
+  // The map is invertable for all r<=outer_boundary as long as 0<a<3b/2.
+  if (map_list.at(f_of_t_b_).func(time)[0][0] <=
+      2.0 / 3.0 * map_list.at(f_of_t_a_).func(time)[0][0]) {
+    ERROR("The map is only invertable if expansion_b < expansion_a*2/3, however"
```

Comment:
Is it possible to check for invertibility at construction time? That way we only allow the construction of maps that won't error halfway through

Reply:
The condition for the map to be invertible is dependent on two FunctionsOfTime. However, these FunctionsOfTime are determined at runtime, as they depend on evolving parameters (e.g. the distance between the horizons during a run), so unfortunately we can't know if the map might become non-invertible at compile time.